core/src/test/java/io/github/tcdl/msb/api/RequesterResponderIT.java

```diff
@@ -61,6 +64,56 @@ public void testResponderServerReceiveMessageSendByRequester() throws Exception
     }
 
     @Test
+    public void testResponderServerReceiveCustomPayloadMessageSendByRequester() throws Exception {
+        String namespace = "test:requester-responder-test-custom-request-received";
+        RequestOptions requestOptions = TestUtils.createSimpleRequestOptions();
+        CountDownLatch requestReceived = new CountDownLatch(1);
+
+        //Create and send request message
+        Requester requester = msbContext.getObjectFactory().createRequester(namespace, requestOptions);
+        Body sentBody = new Body("test:requester-responder-test-body");
+        Payload requestPayload = new Payload.Builder().withBody(sentBody).build();
+
+        Body receivedBody = new Body();
+        msbContext.getObjectFactory().createResponderServer(namespace, requestOptions.getMessageTemplate(), (request, response) -> {
+            MyPayload payload = (MyPayload) request;
+            receivedBody.setBody(payload.getBody().getBody());
+            requestReceived.countDown();
+        }, MyPayload.class)
+                .listen();
+
+        requester.publish(requestPayload);
+
+        assertTrue("Message was not received", requestReceived.await(MESSAGE_TRANSMISSION_TIME, TimeUnit.MILLISECONDS));
+        assertEquals(receivedBody, sentBody);
+    }
+
+    @Test
+    public void testResponderServerReceiveCustomPayloadMessageSendByRequesterAndConvertBody() throws Exception {
+        String namespace = "test:requester-responder-test-custom-request-received-and-converted";
+        RequestOptions requestOptions = TestUtils.createSimpleRequestOptions();
+        CountDownLatch requestReceived = new CountDownLatch(1);
+
+        //Create and send request message
+        Requester requester = msbContext.getObjectFactory().createRequester(namespace, requestOptions);
+        Body sentBody = new Body("test:requester-responder-test-body");
+        Payload requestPayload = new Payload.Builder().withBody(sentBody).build();
+
+        Body receivedBody = new Body();
+        msbContext.getObjectFactory().createResponderServer(namespace, requestOptions.getMessageTemplate(), (request, response) -> {
+            PayloadConverter payloadConverter = msbContext.getObjectFactory().getPayloadConverter();
+            receivedBody.setBody(payloadConverter.getAs(request.getBody(), Body.class).getBody());
```

Comment:
This line is hardly readable. Can we make it more clear with nice variable names?